File: Calcular.py
import PySimpleGUI as sg

# Distâncias pré-definidas em metros
distances = {"5 km": 5000, "10 km": 10000, "15 km": 15000, "Meia-maratona": 21095, "Maratona": 42195}

# Layout da interface gráfica
layout = [
    [sg.Text("Distância"), sg.Combo(list(distances.keys()), key="-DISTANCE-"), sg.Text("Ritmo (min/km)"), sg.InputText(key="-PACE-")],
    [sg.Radio("Calcular Tempo Total e Velocidade Média", "radio_calculation", default=True, key="-TIME_SPEED-"), sg.Radio("Calcular Ritmo e Velocidade Média", "radio_calculation", key="-PACE_SPEED-")],
    [sg.Button("Calcular")],
    [sg.Text("Resultado"), sg.Text("", key="-OUTPUT-")]
]

# Criação da janela
window = sg.Window("Calculadora de Corrida", layout)

while True:
    event, values = window.read()
    if event == sg.WINDOW_CLOSED:
        break
    
    # Obtem a distância e o ritmo do usuário
    distance = distances[values["-DISTANCE-"]]
    pace = float(values["-PACE-"])
    
    # Verifica qual opção de cálculo foi selecionada
    if values["-TIME_SPEED-"]:
        # Calcula o tempo total e a velocidade média
        time = pace * distance / 60
        speed = distance / 1000 / (time / 60)
        
        # Exibe o resultado
        window["-OUTPUT-"].update(f"Tempo Total: {time:.2f} minutos\nVelocidade Média: {speed:.2f} km/h")
        
    elif values["-PACE_SPEED-"]:
        # Calcula o ritmo e a velocidade média
        pace_time = pace / distance * 60
        speed = distance / 1000 / (pace_time / 60)
        
        # Exibe o resultado
        window["-OUTPUT-"].update(f"Ritmo: {pace_time:.2f} minutos/km\nVelocidade Média: {speed:.2f} km/h")

File: Calcular.js
<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Calculadora de Corrida</title>
  </head>
  <body>
    <div>
      <button id="tempo-velocidade-btn">Calcular Tempo total e Velocidade Média</button>
      <button id="pace-velocidade-btn">Calcular Pace e Velocidade Média</button>
      <button id="sair-btn">Sair</button>
    </div>

    <div id="distancia-modal" style="display: none">
      <h3>Escolha uma das distâncias pré-definidas ou insira uma distância qualquer (em Km):</h3>
      <select id="distancia-combo">
        <option value=""></option>
        <option value="5">5 Km</option>
        <option value="10">10 Km</option>
        <option value="15">15 Km</option>
        <option value="21.097">21.097 Km</option>
        <option value="41.195">41.195 Km</option>
      </select>
      <input type="number" id="distancia-input" />
      <button id="distancia-ok-btn">Ok</button>
      <button id="distancia-cancelar-btn">Cancelar</button>
    </div>

    <div id="ritmo-modal" style="display: none">
      <h3>Qual o ritmo desejado (em minutos por Km)?</h3>
      <input type="number" id="ritmo-input" />
      <button id="ritmo-ok-btn">Ok</button>
      <button id="ritmo-cancelar-btn">Cancelar</button>
    </div>

    <div id="pace-modal" style="display: none">
      <h3>Qual o pace desejado (em minutos por Km)?</h3>
      <input type="number" id="pace-input" />
      <button id="pace-ok-btn">Ok</button>
      <button id="pace-cancelar-btn">Cancelar</button>
    </div>

    <div id="resultado-modal" style="display: none">
      <h3>Resultado:</h3>
      <p>Tempo Total: <span id="tempo-total"></span></p>
      <p>Velocidade Média (Km/h): <span id="velocidade-media"></span></p>
    </div>

    <script>
      const distanciaModal = document.getElementById("distancia-modal");
      const distanciaCombo = document.getElementById("distancia-combo");
      const distanciaInput = document.getElementById("distancia-input");
      const distanciaOkBtn = document.getElementById("distancia-ok-btn");
      const distanciaCancelarBtn = document.getElementById("distancia-cancelar-btn");
      const ritmoModal = document.getElementById("ritmo-modal");
      const ritmoInput = document.getElementById("ritmo-input");
      const ritmoOkBtn = document.getElementById("ritmo-ok-btn");
      const ritmoCancelarBtn = document.getElementById("ritmo-cancelar-btn");
      const paceModal = document.getElementById("pace-modal");
      const paceInput = document.getElementById("pace-input");
      const paceOkBtn = document.getElementById("pace-ok-btn");
      const paceCancelarBtn = document.getElementById("pace-cancelar-btn");
      const resultadoModal = document.getElementById("
